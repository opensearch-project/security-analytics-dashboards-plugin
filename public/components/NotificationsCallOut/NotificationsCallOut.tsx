/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';

export const NotificationsCallOut = () => {
  return (
    <>
      <EuiCallOut title="Notifications plugin is not installed" color="danger" iconType="alert">
        <p>
          Install the notifications plugin in order to create and select channels to send out
          notifications.&nbsp;
          <EuiLink href="https://opensearch.org/docs/latest/notifications-plugin/index/" external>
            Learn more
          </EuiLink>
          .
        </p>
      </EuiCallOut>
      <EuiSpacer size="m" />
    </>
  );
};

export default NotificationsCallOut;
