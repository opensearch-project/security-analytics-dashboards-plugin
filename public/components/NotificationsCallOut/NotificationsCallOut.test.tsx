/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import { NotificationsCallOut } from './NotificationsCallOut';

describe('<NotificationsCallOut /> spec', () => {
  it('renders the component', () => {
    const tree = render(<NotificationsCallOut />);
    expect(tree).toMatchSnapshot();
  });
});
