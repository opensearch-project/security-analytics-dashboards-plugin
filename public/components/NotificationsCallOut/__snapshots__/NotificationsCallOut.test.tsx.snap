// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationsCallOut /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="euiCallOut euiCallOut--danger"
      >
        <div
          class="euiCallOutHeader"
        >
          EuiIconMock
          <span
            class="euiCallOutHeader__title"
          >
            Notifications plugin is not installed
          </span>
        </div>
        <div
          class="euiText euiText--small"
        >
          <div
            class="euiTextColor euiTextColor--default"
          >
            <p>
              Install the notifications plugin in order to create and select channels to send out notifications. 
              <a
                class="euiLink euiLink--primary"
                href="https://opensearch.org/docs/latest/notifications-plugin/index/"
                rel="noreferrer"
              >
                Learn more
                EuiIconMock
              </a>
              .
            </p>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--m"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="euiCallOut euiCallOut--danger"
    >
      <div
        class="euiCallOutHeader"
      >
        EuiIconMock
        <span
          class="euiCallOutHeader__title"
        >
          Notifications plugin is not installed
        </span>
      </div>
      <div
        class="euiText euiText--small"
      >
        <div
          class="euiTextColor euiTextColor--default"
        >
          <p>
            Install the notifications plugin in order to create and select channels to send out notifications. 
            <a
              class="euiLink euiLink--primary"
              href="https://opensearch.org/docs/latest/notifications-plugin/index/"
              rel="noreferrer"
            >
              Learn more
              EuiIconMock
            </a>
            .
          </p>
        </div>
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--m"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
